source/agora/script/Engine.d

```diff
@@ -1457,7 +1457,7 @@ unittest
     const Pair kp = Pair.random();
     const Transaction tx;
     const sig = sign(kp, tx);
-    const key_hash = hashFull(kp.V);
+    const key_hash = hashFull(kp.V[]);
```

Comment:
Ditto, why ?

Reply:
These tests fail without slicing. Do not understand why. @AndrejMitrovic do you remember we debugged this together a few weeks back?

Reply:
It's because OP.HASH hashes the first item on the stack, and the first item is a byte array. It's not a fixed-length array. That type info is gone in the script.

Reply:
```D
struct FixedLength
{
    const(ubyte)[] data;
    
    public void computeHash (scope HashDg dg) const @safe nothrow @nogc
    {
        switch (this.data.length)
        {
            // The last value is not included, so it stops at length == 75
            // Keep in sync with `agora.script.Opcodes : OP`
            static foreach (idx; 1 .. 76)
            {
                case idx:
                    hashPart(this.data[0 .. idx + 1], dg);
                    break;
             }
             // Fall back to hashing the length
             default:
                 hashPart(this.data, dg);
        }
    }
}
```

This should do.

Reply:
Using new crypto submodule which supports hashing ubyte[] without length using HashNoLength struct.